The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12527670748/job/34941483190) [(plain)](https://github.com/rust-lang/rust/commit/95aca052b5e6e9fde8b85188e6fe761c848afd75/checks/34941483190/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.887 Building wheels for collected packages: reuse
#13 2.889   Building wheel for reuse (pyproject.toml): started
#13 3.137   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.138   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 3.138   Stored in directory: /tmp/pip-ephem-wheel-cache-awauw9p9/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.141 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.533 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.534 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.058 Collecting virtualenv
#13 4.058 Collecting virtualenv
#13 4.204   Downloading virtualenv-20.28.0-py3-none-any.whl (4.3 MB)
#13 4.271      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 66.0 MB/s eta 0:00:00
#13 4.330 Collecting filelock<4,>=3.12.2
#13 4.333   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.366 Collecting platformdirs<5,>=3.9.1
#13 4.369   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.386 Collecting distlib<1,>=0.3.7
#13 4.397      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 84.8 MB/s eta 0:00:00
#13 4.397      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 84.8 MB/s eta 0:00:00
#13 4.476 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.660 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.28.0
#13 DONE 4.7s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      120768 kB
DirectMap2M:     6170624 kB
DirectMap1G:    12582912 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
WARN: `rust.debug-assertions = true` will prevent downloading CI rustc as alt CI rustc is not currently built with debug assertions.
---
Diff in /checkout/tests/codegen/const-array.rs:2:
 
 #![crate_type = "lib"]
 
-const LUT: [u8; 2] = [
-    1,
-];
-];
+const LUT: [u8; 2] = [1, 1];
 
 // CHECK-LABEL: @decode
 #[no_mangle]
Diff in /checkout/tests/codegen/const-array.rs:14:
     // CHECK-NEXT: icmp
     // CHECK-NEXT: select
     // CHECK-NEXT: ret
-    if i < 2 {
-        LUT[i as usize]
-        2
-    }
-    }
+    if i < 2 { LUT[i as usize] } else { 2 }
 
 
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/tests/codegen/align-byval-alignment-mismatch.rs" "/checkout/tests/codegen/coroutine-debug-msvc.rs" "/checkout/tests/codegen/align-byval.rs" "/checkout/tests/codegen/slice-iter-len-eq-zero.rs" "/checkout/tests/codegen/noalias-refcell.rs" "/checkout/tests/assembly/asm/inline-asm-avx.rs" "/checkout/tests/assembly/asm/avr-types.rs" "/checkout/tests/assembly/asm/loongarch-type.rs" "/checkout/tests/assembly/asm/x86-types.rs" "/checkout/tests/assembly/asm/comments.rs" "/checkout/tests/assembly/asm/msp430-types.rs" "/checkout/tests/assembly/asm/sparc-types.rs" "/checkout/tests/assembly/asm/avr-modifiers.rs" "/checkout/tests/assembly/asm/m68k-types.rs" "/checkout/tests/assembly/asm/arm-modifiers.rs" "/checkout/tests/assembly/asm/powerpc-types.rs" "/checkout/tests/assembly/asm/aarch64-types.rs" "/checkout/tests/assembly/asm/wasm-types.rs" "/checkout/tests/assembly/asm/aarch64-outline-atomics.rs" "/checkout/tests/assembly/asm/s390x-types.rs" "/checkout/tests/assembly/asm/riscv-types.rs" "/checkout/tests/assembly/asm/arm-types.rs" "/checkout/tests/assembly/asm/bpf-types.rs" "/checkout/tests/assembly/asm/x86-modifiers.rs" "/checkout/tests/assembly/asm/aarch64-el2vmsa.rs" "/checkout/tests/assembly/asm/nvptx-types.rs" "/checkout/tests/assembly/asm/mips-types.rs" "/checkout/tests/assembly/asm/hexagon-types.rs" "/checkout/tests/assembly/asm/global_asm.rs" "/checkout/tests/assembly/asm/aarch64-modifiers.rs" "/checkout/tests/assembly/x86_64-sse_crc.rs" "/checkout/tests/assembly/nvptx-arch-link-arg.rs" "/checkout/tests/codegen/loongarch-abi/call-llvm-intrinsics.rs" "/checkout/tests/codegen/loongarch-abi/loongarch64-lp64d-abi.rs" "/checkout/tests/codegen/const-array.rs" "/checkout/tests/codegen/step_by-overflow-checks.rs" "/checkout/tests/assembly/libs/issue-115339-zip-arrays.rs" "/checkout/tests/assembly/x86_64-fortanix-unknown-sgx-lvi-generic-load.rs" "/checkout/tests/assembly/nvptx-arch-emit-asm.rs" "/checkout/tests/assembly/slice-is_ascii.rs" "/checkout/tests/assembly/x86_64-bigint-helpers.rs" "/checkout/tests/assembly/s390x-vector-abi.rs" "/checkout/tests/assembly/panic-unwind-no-uwtable.rs" "/checkout/tests/assembly/wasm_exceptions.rs" "/checkout/tests/assembly/x86_64-no-jump-tables.rs" "/checkout/tests/assembly/simd-intrinsic-mask-reduce.rs" "/checkout/tests/assembly/breakpoint.rs" "/checkout/tests/assembly/x86_64-windows-float-abi.rs" "/checkout/tests/codegen/meta-filecheck/check-prefix.rs" "/checkout/tests/codegen/meta-filecheck/msvc-prefix-bad.rs" "/checkout/tests/codegen/meta-filecheck/no-directives.rs" "/checkout/tests/codegen/meta-filecheck/filecheck-flags.rs" "/checkout/tests/codegen/meta-filecheck/revision-prefix.rs" "/checkout/tests/codegen/wasm_casts_trapping.rs" "/checkout/tests/codegen/call-llvm-intrinsics.rs" "/checkout/tests/codegen/move-before-nocapture-ref-arg.rs" "/checkout/tests/codegen/vecdeque-drain.rs" "/checkout/tests/codegen/maybeuninit-rvo.rs" "/checkout/tests/codegen/float/f16-f128-inline.rs" "/checkout/tests/codegen/float/f16.rs" "/checkout/tests/codegen/float/f128.rs" "/checkout/tests/codegen/default-visibility.rs" "/checkout/tests/codegen/slice-position-bounds-check.rs" "/checkout/tests/codegen/panic-abort-windows.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Sat Dec 28 15:25:56 UTC 2024
  network time: Sat, 28 Dec 2024 15:25:56 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/tests/codegen/const-array.rs
+++ b/tests/codegen/const-array.rs
@@ -1,0 +1,15 @@
+//@ compile-flags: -O
+
+#![crate_type = "lib"]
+
+const LUT: [u8; 2] = [1, 1];
+
+// CHECK-LABEL: @decode
+#[no_mangle]
+pub fn decode(i: u8) -> u8 {
+    // CHECK: start:
+    // CHECK-NEXT: icmp
+    // CHECK-NEXT: select
+    // CHECK-NEXT: ret
+    if i < 2 { LUT[i as usize] } else { 2 }
+}

